File: backend/app/dto/responses/wallet.go
package responses

type WalletAnalysis struct {
	Address           string        `json:"address"`
	Chain             string        `json:"chain"`
	TotalTransactions int           `json:"total_transactions"`
	ProfitConsistency float64       `json:"profit_consistency"`
	WinRate           float64       `json:"win_rate"`
	RiskExposure      float64       `json:"risk_exposure"`
	EntryTiming       float64       `json:"entry_timing"`
	TokenQuality      float64       `json:"token_quality"`
	TradeDiscipline   float64       `json:"trade_discipline"`
	FinalScore        float64       `json:"final_score"`
	Recommendation    string        `json:"recommendation"`
	AiInsight         string        `json:"ai_insight,omitempty"`
	TradedTokens      []TradedToken `json:"traded_tokens,omitempty"`
}

// TradedToken represents a token the wallet traded, with PnL data.
type TradedToken struct {
	ContractAddress string  `json:"contract_address"`
	Symbol          string  `json:"symbol"`
	Chain           string  `json:"chain"`
	PnlPercent      float64 `json:"pnl_percent"`
	BuyPrice        float64 `json:"buy_price"`
	ExitPrice       float64 `json:"exit_price"`
	Status          string  `json:"status"` // "realized" or "unrealized"
}

File: frontend/types/wallet.ts
export type TradedToken = {
  contract_address: string;
  symbol: string;
  chain: string;
  pnl_percent: number;
  buy_price: number;
  exit_price: number;
  buy_time: string;
  exit_time?: string;
  status: "realized" | "unrealized";
};

export type Condition = {
  id: string;
  label: string;
  description: string;
  type: string;
  field: string;
  operator: string;
  value: number;
};

export type WalletAnalysis = {
  address: string;
  chain: string;
  total_transactions: number;
  profit_consistency: number;
  win_rate: number;
  risk_exposure: number;
  entry_timing: number;
  token_quality: number;
  trade_discipline: number;
  final_score: number;
  recommendation: "full_follow" | "conditional_follow" | "avoid";
  ai_insight?: string;
  ai_insight_tone?: string;
  ai_insight_prompt?: string;
  traded_tokens?: TradedToken[];
  conditions?: Condition[];
};
